### ISSUE TITLE
More unique clientId

### ISSUE THREAD

**Issue by vanshg** [eclipse/paho.mqtt.android#120]
Right now, with the current implementation of combining a static String with the current time, there is chance of users having overlapping clientIds, especially in an application with many users on the same channel.

Maybe use the UUID of the Android device, or generate a random String and use that.

For example, `String clientId = UUID.randomUUID().toString()`

That way 1) the Client ID is more unique and 2) There is less chance for overlap of client ids

**Comment by jpwsutton** [comment]
Hi, Are you referring to just this in the Example? (https://github.com/eclipse/paho.mqtt.android/blob/develop/paho.mqtt.android.example/src/main/java/paho/mqtt/java/example/PahoExampleActivity.java#L79). That would be an improvement for those trying out the Example project, though I'd imagine that the chances of multiple people starting the Example App at the same time and causing a clash is quite unlikely.

**Comment by vanshg** [comment]
Nope, I'm talking about the `MqttClient.generateClientId` and `MqttAsyncClient.generateClientId` methods. 

```
/**
	 * Returns a randomly generated client identifier based on the the fixed prefix (paho)
	 * and the system time.
	 * <p>When cleanSession is set to false, an application must ensure it uses the
	 * same client identifier when it reconnects to the server to resume state and maintain
	 * assured message delivery.</p>
	 * @return a generated client identifier
	 * @see MqttConnectOptions#setCleanSession(boolean)
	 */
	public static String generateClientId() {
		//length of nanoTime = 15, so total length = 19  < 65535(defined in spec) 
		return CLIENT_ID_PREFIX + System.nanoTime();
	}
```

This method claims that it creates a random client id, but it's not really random, it's based off a fixed prefix ("paho") and the current time, and if there's an application with many users, users may end up getting the same client id

I guess I really should've opened this issue in the paho.mqtt.java repo

**Comment by jpwsutton** [comment]
Ah, that would explain why I could only find that in the example project!
Yes, that is an issue that should probably be improved, though Ideally everyone should be setting their own client ID anyway.

As Java.util.uuid is not available in Java 1.4.2, we will have to wait until after the 1.3.0 release before we can make this change. (After 1.3.0 we will be splitting off a dedicated 1.4.2 compatible branch, and raising the minimum source compatibility to 1.7+ in the main develop and master branches)

**Comment by jpwsutton** [comment]
This issue was moved to eclipse/paho.mqtt.java#265
